File: enums.go
package main

const (
	actionUnknown                   = 0
	actionDel                       = 1
	actionRead                      = 2
	actionWrite                     = 4
	actionOpen                      = 8
	actionMkdir                     = 16
	actionNFSMountSucc              = 32
	actionNFSMountFail              = 64
	actionRename                    = 128
	actionLoginFail                 = 256
	actionLoginSucc                 = 512
	actionLogout                    = 1024
	actionNFSUmount                 = 2048
	actionCopy                      = 4096
	actionMove                      = 8192
	actionAdd                       = 16384
	actionAuthFail                  = 32768
	actionAuthOk                    = 65536
	actionTrashRecovery             = 131072
	actionAddTranscode              = 262144
	actionDelTranscode              = 524288
	actionUpdateTranscode           = 1048576
	actionWatermark                 = 2097152
	actionRotate                    = 4194304
	actionAddThumbnail              = 8388608
	actionAddFiling                 = 16777216
	actionUpdateFiling              = 33554432
	actionDeleteFiling              = 67108864
	actionPauseFiling               = 134217728
	actionStopFiling                = 268435456
	actionEject                     = 536870912
	actionEncrypt                   = 1073741824
	actionDecrypt                   = 2147483648
	actionConnActionCreateShareLink = 4294967296
	actionConnActionCompress        = 8589934592
	actionConnActionExtract         = 17179869184
	actionConnActionCloudconvert    = 34359738368
	actionAny                       = 68719476735
)

var actionMap = map[int]string{
	actionUnknown:                   `Unknown`,
	actionDel:                       `Del`,
	actionRead:                      `Read`,
	actionWrite:                     `Write`,
	actionOpen:                      `Open`,
	actionMkdir:                     `Mkdir`,
	actionNFSMountSucc:              `NFSMountSucc`,
	actionNFSMountFail:              `NFSMountFail`,
	actionRename:                    `Rename`,
	actionLoginFail:                 `LoginFail`,
	actionLoginSucc:                 `LoginSucc`,
	actionLogout:                    `Logout`,
	actionNFSUmount:                 `NFSUmount`,
	actionCopy:                      `Copy`,
	actionMove:                      `Move`,
	actionAdd:                       `Add`,
	actionAuthFail:                  `AuthFail`,
	actionAuthOk:                    `AuthOk`,
	actionTrashRecovery:             `TrashRecovery`,
	actionAddTranscode:              `AddTranscode`,
	actionDelTranscode:              `DelTranscode`,
	actionUpdateTranscode:           `UpdateTranscode`,
	actionWatermark:                 `Watermark`,
	actionRotate:                    `Rotate`,
	actionAddThumbnail:              `AddThumbnail`,
	actionAddFiling:                 `AddFiling`,
	actionUpdateFiling:              `UpdateFiling`,
	actionDeleteFiling:              `DeleteFiling`,
	actionPauseFiling:               `PauseFiling`,
	actionStopFiling:                `StopFiling`,
	actionEject:                     `Eject`,
	actionEncrypt:                   `Encrypt`,
	actionDecrypt:                   `Decrypt`,
	actionConnActionCreateShareLink: `ConnActionCreateShareLink`,
	actionConnActionCompress:        `ConnActionCompress`,
	actionConnActionExtract:         `ConnActionExtract`,
	actionConnActionCloudconvert:    `ConnActionCloudconvert`,
	actionAny:                       `Any`,
}

const (
	serviceUnknown = 0
	serviceSamba   = 1
	serviceFtp     = 2
	serviceHttp    = 4
	serviceNfs     = 8
	serviceAfp     = 16
	serviceTelnet  = 32
	serviceSsh     = 64
	serviceIscsi   = 128
	serviceRadius  = 256
	serviceVpn     = 512
	serviceHttps   = 1024
	serviceAny     = 2047
)

var serviceMap = map[int]string{
	serviceUnknown: `Unknown`,
	serviceSamba:   `Samba`,
	serviceFtp:     `Ftp`,
	serviceHttp:    `Http`,
	serviceNfs:     `Nfs`,
	serviceAfp:     `Afp`,
	serviceTelnet:  `Telnet`,
	serviceSsh:     `Ssh`,
	serviceIscsi:   `Iscsi`,
	serviceRadius:  `Radius`,
	serviceVpn:     `Vpn`,
	serviceHttps:   `Https`,
	serviceAny:     `Any`,
}

File: tools/enums.js

let enums = {
     ACTION: {
      UNKNOWN: 0,
      DEL: Math.pow(2, 0),
      READ: Math.pow(2, 1),
      WRITE: Math.pow(2, 2),
      OPEN: Math.pow(2, 3),
      MKDIR: Math.pow(2, 4),
      N_F_S_MOUNT_SUCC: Math.pow(2, 5),
      N_F_S_MOUNT_FAIL: Math.pow(2, 6),
      RENAME: Math.pow(2, 7),
      LOGIN_FAIL: Math.pow(2, 8),
      LOGIN_SUCC: Math.pow(2, 9),
      LOGOUT: Math.pow(2, 10),
      N_F_S_UMOUNT: Math.pow(2, 11),
      COPY: Math.pow(2, 12),
      MOVE: Math.pow(2, 13),
      ADD: Math.pow(2, 14),
      AUTH_FAIL: Math.pow(2, 15),
      AUTH_OK: Math.pow(2, 16),
      TRASH_RECOVERY: Math.pow(2, 17),
      ADD_TRANSCODE: Math.pow(2, 18),
      DEL_TRANSCODE: Math.pow(2, 19),
      UPDATE_TRANSCODE: Math.pow(2, 20),
      WATERMARK: Math.pow(2, 21),
      ROTATE: Math.pow(2, 22),
      ADD_THUMBNAIL: Math.pow(2, 23),
      ADD_FILING: Math.pow(2, 24),
      UPDATE_FILING: Math.pow(2, 25),
      DELETE_FILING: Math.pow(2, 26),
      PAUSE_FILING: Math.pow(2, 27),
      STOP_FILING: Math.pow(2, 28),
      EJECT: Math.pow(2, 29),
      ENCRYPT: Math.pow(2, 30),
      DECRYPT: Math.pow(2, 31),
      CONN_ACTION_CREATE_SHARE_LINK: Math.pow(2, 32),
      CONN_ACTION_COMPRESS: Math.pow(2, 33),
      CONN_ACTION_EXTRACT: Math.pow(2, 34),
      CONN_ACTION_CLOUDCONVERT: Math.pow(2, 35),
      ANY: Math.pow(2, 36) - 1
    },
    SERVICE: {
      UNKNOWN: 0,
      SAMBA: 1 << 0,
      FTP: 1 << 1,
      HTTP: 1 << 2,
      NFS: 1 << 3,
      AFP: 1 << 4,
      TELNET: 1 << 5,
      SSH: 1 << 6,
      ISCSI: 1 << 7,
      RADIUS: 1 << 8,
      VPN: 1 << 9,
      HTTPS: 1 << 10,
      ANY: (1 << 11) - 1
      // ANY:     ((1 << 9) - 1) ^ (1 << 3)
    },
}

String.prototype.toPascalCase = function() {
    const words = this.match(/[a-z]+/gi);
    if (!words) return "";
    return words
        .map(function(word) {
            return word.charAt(0).toUpperCase() + word.substr(1).toLowerCase();
        })
        .join("");
};

console.log("package main");
console.log("");

console.log("const(");
for(let x in enums.ACTION) {
    console.log("    ", "action" + x.toPascalCase(), "=", enums.ACTION[x]);
}
console.log(")");

console.log("");

console.log("var actionMap = map[int]string{");
for(let x in enums.ACTION) {
    console.log("    ", "action"+x.toPascalCase(), ":", "`" + x.toPascalCase() + "`,");
}
console.log("}");

console.log("");

console.log("const(");
for(let x in enums.SERVICE) {
    console.log("    ", "service" + x.toPascalCase(), "=", enums.SERVICE[x]);
}
console.log(")");

console.log("");

console.log("var serviceMap = map[int]string{");
for(let x in enums.SERVICE) {
    console.log("    ", "service"+x.toPascalCase(), ":", "`" + x.toPascalCase() + "`,");
}
console.log("}");